restapi/src/main/java/io/stargate/web/docsapi/exception/ErrorCodeRuntimeException.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.web.docsapi.exception;
+
+import java.util.Objects;
+import javax.ws.rs.core.Response;
+
+/** {@link RuntimeException} that carries an {@link ErrorCode}. */
+public class ErrorCodeRuntimeException extends RuntimeException {
+
+  /** Dedicated {@link ErrorCode}. */
+  private final ErrorCode errorCode;
+
+  /** ErrorCode only constructor. */
+  public ErrorCodeRuntimeException(ErrorCode errorCode) {
+    this.errorCode = errorCode;
+  }
+
+  /** @see Exception#Exception(String) */
+  public ErrorCodeRuntimeException(ErrorCode errorCode, String message) {
+    super(message);
+    this.errorCode = errorCode;
+  }
+
+  /** @see Exception#Exception(String, Throwable) */
+  public ErrorCodeRuntimeException(ErrorCode errorCode, String message, Throwable cause) {
+    super(message, cause);
+    this.errorCode = errorCode;
+  }
+
+  /** Pipe to {@link ErrorCode#toResponse()} with the exception message. */
+  public Response getResponse() {
+    return errorCode.toResponse(getLocalizedMessage());
+  }
+
+  /** Pipe to {@link ErrorCode#toResponseBuilder(String)} with the exception message. */
+  public Response.ResponseBuilder getResponseBuilder() {
+    return errorCode.toResponseBuilder(getLocalizedMessage());
+  }
+
+  public ErrorCode getErrorCode() {
+    return errorCode;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    ErrorCodeRuntimeException that = (ErrorCodeRuntimeException) o;
+    return errorCode == that.errorCode;
+  }
```

Comment:
What is the purpose of this particular way of doing `equals`? Exceptions can differ by origin/stack trace as well as by error code :thinking:

Reply:
Yea, this will be removed.. Thanks..